rust/abacus-base/src/contract_sync/outbox.rs

```diff
@@ -87,28 +101,45 @@ where
                     continue;
                 }
 
-                let candidate = from + chunk_size;
-                let to = min(tip, candidate);
+                // Index the chunk_size, capping at the tip.
+                let to = min(tip, from + chunk_size);
 
-                let sorted_messages = indexer.fetch_sorted_messages(from, to).await?;
+                let mut sorted_messages = indexer.fetch_sorted_messages(from, to).await?;
 
-                debug!(
+                info!(
                     from = from,
                     to = to,
                     message_count = sorted_messages.len(),
                     "[Messages]: indexed block heights {from}...{to}"
                 );
 
-                // If no messages found, update last seen block and next height
-                // and continue
+                // Get the latest known leaf index. All messages whose indices are <= this index
+                // have been stored in the DB.
+                let last_leaf_index: OptLatestLeafIndex = db.retrieve_latest_leaf_index()?.into();
```

Comment:
Sorry realizing I didn't respond here. I like the idea but I'm wary of DB changes. I'm realizing my "Backward compatibility" section was actually inaccurate because of this value that's persisted-- this will prevent any re-indexing of messages that the db has already indexed, which is quite nice, because of that filter afterward. Changing this would probably make the backward compatibility slightly less easy. You ok if I skip this?

Reply:
Yup totally reasonable, please go ahead